### PR TITLE
[TFLM] Added support for optimized quantize kernel for CEVA-BX1 and CEVA-SP500 cores

### PR DESCRIPTION
**Pull request by yair-ehrenwald**
see issue https://github.com/tensorflow/tensorflow/issues/45607

Started with quantize but more to follow
Also some fixes for CEVA makefiles in this PR.

### CODE REVIEW

**Comment by gbaned** [comment]
@yair-ehrenwald  Can you please resolve conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

File: tensorflow/lite/micro/tools/make/templates/ceva_SP500/ceva_app_makefile.tpl [hunk 1]
```diff
@@ -0,0 +1,68 @@
+
+TARGET_TOOLCHAIN_ROOT := /home/yaire/CEVA-ToolBox/V20/SensPro
```
Comment (njeffrie) [inline tensorflow/lite/micro/tools/make/templates/ceva_SP500/ceva_app_makefile.tpl:2]: Seems like having a hardcoded path is likely an error? I notice this is also present in the other ceva template file as well.

File: tensorflow/lite/micro/tools/make/targets/ceva/CEVA_SP500_TFLM.ld [hunk 2]
```diff
@@ -0,0 +1,219 @@
+OUTPUT(a.elf)
```
Comment (njeffrie) [inline tensorflow/lite/micro/tools/make/targets/ceva/CEVA_SP500_TFLM.ld:1]: Please add copyright notice (see other ld files for reference)

File: tensorflow/lite/micro/kernels/ceva/quantize.cc [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  TFLITE_DCHECK(node->user_data != nullptr);
+  OpData* data = static_cast<OpData*>(node->user_data);
+
+  const TfLiteEvalTensor* input = tflite::micro::GetEvalInput(context, node, 0);
+  TfLiteEvalTensor* output = tflite::micro::GetEvalOutput(context, node, 0);
+
+  if (input->type == kTfLiteFloat32) {
+    switch (output->type) {
+      case kTfLiteInt8: {
+#ifdef ORIGINAL_IMPLEMENTATION
```
Comment (njeffrie) [inline tensorflow/lite/micro/kernels/ceva/quantize.cc:112]: I'm curious how this is different compared with running EvalQuantizeReference. Perhaps this is some dead test code?

**Review by njeffrie — CHANGES_REQUESTED** [review]
Please run:
clang-format --style=google tensorflow/lite/micro/kernels/ceva/CEVA_TFLM_lib.h

or 

tensorflow/lite/micro/tools/make/downloads/pigweed/pw_presubmit/py/pw_presubmit/format_code.py --fix /workspace/tensorflow/lite/micro/kernels/ceva/CEVA_TFLM_lib.h

These should be equivalent (at least in my experience).

**Comment by yair-ehrenwald** [comment]
Closing this, opened a new PR https://github.com/tensorflow/tensorflow/pull/47781

**Pull request closed without merging**